java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java

```diff
@@ -17,12 +18,11 @@
     public EpcisEventContainer() {
     }
 
-    public EpcisEventContainer(final List<EpcisEvent> events) {
-        this.events = events;
+    public EpcisEventContainer(final List<? extends EpcisEvent> events) {
+        this.events = new ArrayList<>(events);
```

Comment:
It seems like an improper usage of using generics in this case.
See also http://stackoverflow.com/questions/1292109/generics-get-and-put-rule

Why not do something like this:
```
  public class EventContainer {

        public List<EpcisEvent> events;

        public EventContainer() {
            events = new ArrayList<>();
        }

        public EventContainer(List<EpcisEvent> events) {
            this.events = events;
        }

        public EventContainer(EpcisEvent epcisEvent) {
            this(Arrays.asList(epcisEvent));
        }

        public EventContainer add(List<ObjectEvent> events) {
            events.addAll(events);
        }
    }

    private void foo() {
        List<ObjectEvent> list = new ArrayList<>();
        EventContainer eventsContainer = new EventContainer().add(list);
    }
```